src/patternfly/components/Button/styles.scss

```diff
@@ -290,6 +312,71 @@
     }
   }
 
+  // Link and Action Buttons
+  &.pf-m-link,
+  &.pf-m-action {
+    background-color: transparent;
+    box-shadow: none;
+
+    &:hover,
+    &.pf-m-hover,
+    &:active,
+    &.pf-m-active,
+    &:focus,
+    &.pf-m-focus {
+      background-color: transparent;
+      box-shadow: none;
+    }
+  }
+
+  // Link Buttons
+  &.pf-m-link {
+    display: inline-flex;
+    padding: var(--pf-c-button--m-link--PaddingTop) var(--pf-c-button--m-link--PaddingRight) var(--pf-c-button--m-link--PaddingBottom) var(--pf-c-button--m-link--PaddingLeft);
+    color: var(--pf-c-button--m-link--Color);
+
+    > * {
```

Comment:
If we want to add this it should be at the level of the btn and not a modifier. Since the designers have not define how they want an icon withing a btn to look, I'd remove this scenario from this PR

Reply:
Does this count as the design example? Or are we looking for more detail?

<img width="321" alt="screen shot 2018-06-05 at 12 53 59 pm" src="https://user-images.githubusercontent.com/5385435/40992468-e716bcec-68c4-11e8-8d5b-f05a4715f695.png">